pyscf/lib/dft/libxc_itrf.c

```diff
@@ -362,6 +363,124 @@ static void _eval_xc(xc_func_type *func_x, int spin, int np,
         }
 }
 
+typedef struct {
+        int id;
+        double *params;
+        size_t size;
+} ext_params;
+
+static ext_params *custom_ext_params = NULL;
+static int custom_ext_params_count = 0;
+
+int set_ext_params(int id, double *params, size_t size)
+{
+        // Update existing entry if exists
+        for (int i = 0; i < custom_ext_params_count; i++) {
+                ext_params *p = &custom_ext_params[i];
+                if (p->id == id) {
+                        if (p->size != size) {
+                                fprintf(stderr, "Inconsistent parameter size. Expected %lu but got %lu\n", p->size, size);
```

Comment:
"Parameter size" is misleading, say "number of parameters"

Reply:
Fixed